apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -639,70 +643,90 @@ export class ApplicationController {
 
     const templateId = existingApplication.typeId as ApplicationTypes
     const template = await getApplicationTemplateByTypeId(templateId)
+    return withCodeOwner(template.codeOwner, async () => {
```

Comment:
You could create a utility function, which assigns codeowner and adds logging context. Invoked like this:

```ts
return withApplicationInfo(template, existingApplication, () => {
  // ...
})
```

Where withApplicationInfo does something like this:

```ts
function withApplicationInfo(template, application, callback) {
  const context = {
    templateId: template.type,
    applicationId: application.id
  }
  return withCodeOwner(template.codeOwner, () =>
    withLoggingContext(context, callback)
  )
}
```